components/datetime/src/format/datetime.rs

```diff
@@ -1094,7 +1096,7 @@ mod tests {
         use icu_calendar::DateTime;
         use icu_provider::prelude::*;
 
-        let locale = "en-u-ca-gregory".parse().unwrap();
+        let locale = "en".parse().unwrap();
```

Comment:
Nit optional: avoid changing the test

Reply:
`-u-ca-gregory` doesn't parse as a `DataLocale` anymore, and that's intended. The only way to currently get non-`-u-sd` subtags into a `DataLocale`, while we don't have preferences, is to convert from a `Locale`. This subtag was previously ignored by this test anyway.